src/zenml/orchestrators/context_utils.py

```diff
@@ -49,3 +51,33 @@ def add_stack_as_metadata_context(
     for k, v in stack_dict.items():
         c_property = context.properties[k]  # type:ignore[attr-defined]
         c_property.field_value.string_value = v
+
+
+def add_pydantic_object_as_metadata_context(
+    obj: "BaseModel",
+    context: "pipeline_pb2.ContextSpec",  # type: ignore[valid-type]
+) -> None:
+    """
+
+    Args:
+        obj: an instance of a pydantic object
+        context: a context proto message within a pipeline node
+    """
+    context.type.name = (  # type: ignore[attr-defined]
+        obj.__repr_name__().lower()
+    )
+    # Setting the name of the context
+    name = str(hash(obj.json(sort_keys=True)))
```

Comment:
How about a try catch for this if obj.json() fails? Or even the hash -> Although technically that should not be possible